subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/checkstyle/CheckstylePluginVersionIntegrationTest.groovy

```diff
@@ -147,6 +146,8 @@ class CheckstylePluginVersionIntegrationTest extends MultiVersionIntegrationSpec
         file("build/reports/checkstyle/main.xml").assertContents(containsClass("org.gradle.Class1"))
     }
 
+    // on parallel executer, there might be checkstyleMain/checkstyleTest running in parallel, resulting in non-deterministic result
+    @IgnoreIf({ GradleContextualExecuter.parallel })
```

Comment:
It seems like we should run `checkstyleMain` only instead of check then? I also don't understand how the two tasks can run in parallel, given that they are in the same project.

Reply:
Like this: https://ge.gradle.org/s/4ktjtercsrabw/tests/:code-quality:parallelIntegTest/org.gradle.api.plugins.quality.checkstyle.CheckstylePluginVersionIntegrationTest/can%20suppress%20console%20output%20%5B7.0%5D?anchor=e30&focused-execution=1&top-execution=1#L18

That's a very good point... without `--continue`, the build should fail at first failed task, right?

Reply:
> That's a very good point... without --continue, the build should fail at first failed task, right?

That is my understanding.